File: backend/src/Naheulbook.Web/Responses/AptitudeResponse.cs
using JetBrains.Annotations;

namespace Naheulbook.Web.Responses;

[PublicAPI]
public class AptitudeResponse
{
    public Guid Id { get; set; }
    public int Roll { get; set; }
    public string Type { get; set; } = null!;
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public string Effect { get; set; } = null!;
}

File: frontend/src/app/api/responses/aptitude-response.ts
export type AptitudeResponse = {
    id: string;
    name: string;
    roll: number;
    type: string;
    description: string;
    effect: string;
}

export type AptitudeGroupResponse = {
    id: string;
    name: string;
    aptitudes: AptitudeResponse[];
}


export type SummaryAptitudeGroupResponse = {
    id: string;
    name: string;
}

export type AptitudeGroupsResponse = {
    aptitudeGroups: SummaryAptitudeGroupResponse[];
}
